### PR TITLE
Read hostURL from the context

### PR DESCRIPTION
**Pull request by harshitav-docusign**
---
### When submitting a connector, please make sure that you follow the requirements below, otherwise your PR might be rejected. We want to make you have a well-built connector, a smooth certification experience, and your users are happy :) 

If this is your first time submitting to GitHub and you need some help, please sign up for this [session](https://forms.office.com/pages/responsepage.aspx?id=KtIy2vgLW0SOgZbwvQuRaXDXyCl9DkBHq4A2OG7uLpdUMTFJWFFGVUxBNUFZQjZWRUdaOE5BMFkwNS4u). 

- [ ] I attest that the connector doesn't exist on the Power Platform today. I've verified by checking the pull requests in GitHub and by searching for the connector on the platform or in the documentation. 
- [ ] I attest that the connector works and I verified by deploying and testing all the operations. 
- [ ] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [ ] I attest that I have added response schemas to my actions, unless the response schema is dynamic. 
- [ ] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [ ] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

If you are an Independent Publisher, you must also attest to the following to ensure a smooth publishing process:
- [ ] I have named this PR after the pattern of "Connector Name (Independent Publisher)" ex: HubSpot Marketing (Independent Publisher)
- [ ] Within this PR markdown file, I have pasted screenshots that show: 3 unique operations (actions/triggers) working within a Flow. This can be in one flow or part of multiple flows. For each one of those flows, I have pasted in screenshots of the Flow succeeding. 
- [ ] Within this PR markdown file, I have pasted in a screenshot from the Test operations section within the Custom Connector UI.
- [ ] If the connector uses OAuth, I have provided detailed steps on how to create an app in the readme.md.

### CODE REVIEW

File: certified-connectors/DocuSign/script.csx [hunk 1]
```diff
@@ -1471,13 +1464,46 @@ public class Script : ScriptBase
...
+      : host.Contains("stage") ?
+        "https://apps-s.docusign.com"
+      : host.Contains(".mil") ?
+        "https://app.docusign.mil"
+      : "https://app.docusign.com";
+
+    return applicationURL;
+  }
+
+  private string GetHostURL()
```
Comment (sriharip-docusign) [inline certified-connectors/DocuSign/script.csx]: consider renaming to `GetAccountServerBaseUri`

File: certified-connectors/DocuSign/script.csx [hunk 2]
```diff
@@ -1471,13 +1464,46 @@ public class Script : ScriptBase
...
+        "/documents/details/" + envelope["envelopeId"]
+      : uriBuilder.Uri.ToString().Contains("stage") ?
+        "/send/documents/details/" + envelope["envelopeId"]
+      : "/documents/details/" + envelope["envelopeId"];
+    var envelopeUrl = GetApplicationURL() + envelopePath;
 
     return envelopeUrl;
   }
 
+  private string GetApplicationURL()
```
Comment (sriharip-docusign) [inline certified-connectors/DocuSign/script.csx]: consider renaming to `GetDocusignApiBaseUri`

File: certified-connectors/DocuSign/script.csx [hunk 3]
```diff
@@ -1471,13 +1464,46 @@ public class Script : ScriptBase
   private string GetEnvelopeUrl(JToken envelope)
   {
     var uriBuilder = new UriBuilder(this.Context.Request.RequestUri);
-    var envelopeUrl = uriBuilder.Uri.ToString().Contains("demo") ?
-      "https://apps-d.docusign.com/send/documents/details/" + envelope["envelopeId"] :
-      "https://app.docusign.com/documents/details/" + envelope["envelopeId"];
+    var envelopePath = uriBuilder.Uri.ToString().Contains("demo") ?
+        "/send/documents/details/" + envelope["envelopeId"]
+      : uriBuilder.Uri.ToString().Contains(".mil") ?
+        "/documents/details/" + envelope["envelopeId"]
```
Comment (sriharip-docusign) [inline certified-connectors/DocuSign/script.csx]: consider simplifying this
* the demo and stage case looks identical
* .mil and the default case looks identical

**Comment by sriharip-docusign** [comment]
@harshitav-docusign , is there any change needed in the swagger file ?

File: certified-connectors/DocuSign/script.csx (hunk 3 above)
Comment (harshitav-docusign) [reply certified-connectors/DocuSign/script.csx]: Done

File: certified-connectors/DocuSign/script.csx (hunk 2 above)
Comment (harshitav-docusign) [reply certified-connectors/DocuSign/script.csx]: Done

File: certified-connectors/DocuSign/script.csx (hunk 1 above)
Comment (harshitav-docusign) [reply certified-connectors/DocuSign/script.csx]: Done

File: certified-connectors/DocuSign/script.csx [hunk 4]
```diff
@@ -1471,13 +1464,42 @@ public class Script : ScriptBase
...
+    "/send/documents/details/" : "/documents/details/";
+    var envelopeUrl = GetDocusignApiBaseUri() + basePath + envelope["envelopeId"];
 
     return envelopeUrl;
   }
 
+  private string GetDocusignApiBaseUri()
+  {
+    var host = this.Context.Request.RequestUri.Host.ToLower();
+    var applicationURL = host.Contains("demo") ?
```
Comment (sriharip-docusign) [inline certified-connectors/DocuSign/script.csx]: rename variable

File: certified-connectors/DocuSign/script.csx [hunk 5]
```diff
@@ -1471,13 +1464,42 @@ public class Script : ScriptBase
...
+        "https://app.docusign.mil"
+      : "https://app.docusign.com";
+
+    return applicationURL;
+  }
+
+  private string GetAccountServerBaseUri()
+  {
+    var host = this.Context.Request.RequestUri.Host.ToLower();
+    var hostURL = host.Contains("demo") ?
```
Comment (sriharip-docusign) [inline certified-connectors/DocuSign/script.csx]: rename variable accordingly

**Comment by harshitav-docusign** [comment]
> @harshitav-docusign , is there any change needed in the swagger file ?

@sriharip-docusign There is also a place in the swagger file where we are hardcoding the host url. I will have to talk to SreeVishnu from the Microsoft India team on how to read it from the context. I will keep this PR open till tomorrow.

File: certified-connectors/DocuSign/script.csx [hunk 6]
```diff
@@ -1471,13 +1464,42 @@ public class Script : ScriptBase
   private string GetEnvelopeUrl(JToken envelope)
   {
     var uriBuilder = new UriBuilder(this.Context.Request.RequestUri);
-    var envelopeUrl = uriBuilder.Uri.ToString().Contains("demo") ?
-      "https://apps-d.docusign.com/send/documents/details/" + envelope["envelopeId"] :
-      "https://app.docusign.com/documents/details/" + envelope["envelopeId"];
+
+    var basePath = uriBuilder.Uri.ToString().Contains("demo") || uriBuilder.Uri.ToString().Contains("stage") ?
```
Comment (sriharip-docusign) [inline certified-connectors/DocuSign/script.csx]: i would just call it as path instead of basePath as the base is moved out to the method

**Review by sriharip-docusign — APPROVED** [review]

**Comment by harshitav-docusign** [comment]
> > @harshitav-docusign , is there any change needed in the swagger file ?
> 
> @sriharip-docusign There is also a place in the swagger file where we are hardcoding the host url. I will have to talk to SreeVishnu from the Microsoft India team on how to read it from the context. I will keep this PR open till tomorrow.

@sriharip-docusign Sreevishnu informed that, the Host URL in the swagger file need not be changed. The serviceURL value will override the Host URL in the swagger file. I am going ahead and merging this PR.

**Pull request merged**